applications/ULFapplication/custom_processes/find_triple_point.h

```diff
@@ -105,117 +108,40 @@ namespace Kratos
     ///@name Operations
     ///@{
 
+
     void FindTriplePoint2D(ModelPart& ThisModelPart)
     {
       KRATOS_TRY
+
+      int is_free = 0;
+      int is_struct = 0;
+      int is_lagin = 0;
       
-      double mean_ystr = 0.0;
-      ///////double xmin = 0.0;
-      ///////double xmax = 0.0;
-      int num_istruct = 0;
-      //FIRST STEP: find IS_STRUCTURE nodes and their mean y coordinate, min and max x coordinate
-      for(ModelPart::NodesContainerType::iterator im = ThisModelPart.NodesBegin() ; im != ThisModelPart.NodesEnd() ; ++im)
+      for(ModelPart::NodesContainerType::iterator im = ThisModelPart.NodesBegin() ; im != ThisModelPart.NodesEnd() ; im++)
       {
 	  if (im->FastGetSolutionStepValue(IS_STRUCTURE) != 0.0)
 	  {
-	      mean_ystr += im->Y();
-	      num_istruct++;
-	  }
-      }
-      mean_ystr /= num_istruct;
-      
-      double dist = 0.0;
-      double eps_tol = 1.0E-6;
-      //SECON STEP: correct the flags
-      for(ModelPart::NodesContainerType::iterator im = ThisModelPart.NodesBegin() ; im != ThisModelPart.NodesEnd() ; ++im)
-      {
-	  if (im->FastGetSolutionStepValue(IS_BOUNDARY) != 0.0 && im->FastGetSolutionStepValue(IS_LAGRANGIAN_INLET) < 1e-15)
-	  {
-	      dist = sqrt((im->Y() - mean_ystr)*(im->Y() - mean_ystr));
-	      if (dist < eps_tol)
-	      {
-		  im->FastGetSolutionStepValue(IS_FREE_SURFACE) = 0.0;
-		  im->FastGetSolutionStepValue(IS_INTERFACE) = 0.0;
-		  im->FastGetSolutionStepValue(IS_STRUCTURE) = 1.0;
-	      }
-	      else
-	      {
-		  im->FastGetSolutionStepValue(IS_FREE_SURFACE) = 1.0;
-		  im->FastGetSolutionStepValue(IS_INTERFACE) = 1.0;
-		  im->FastGetSolutionStepValue(TRIPLE_POINT) = 0.0;
-		  im->FastGetSolutionStepValue(IS_STRUCTURE) = 0.0;		  
-	      }
-	  }
-      }
-      
-      
-      //THIRD STEP: apply TRIPLE_POINT flag
-      double x_mean = 0.0;
-      double num_tp = 0.0;
-      for(ModelPart::NodesContainerType::iterator im = ThisModelPart.NodesBegin() ; im != ThisModelPart.NodesEnd() ; ++im)
-      {
-	  if (im->FastGetSolutionStepValue(IS_BOUNDARY) != 0.0)
-	  {
+	      is_free = 0;
+	      is_struct = 0;
+	      is_lagin = 0;
 	      WeakPointerVector< Node<3> >& neighb = im->GetValue(NEIGHBOUR_NODES);
-	      double is_free = 0.0;
-	      double is_struct = 0.0;
-	      double is_lagin = 0.0;
-	      ///////int neighnum = 0;
-	      
 	      for (unsigned int i = 0; i < neighb.size(); i++)
 	      {
 		  if (neighb[i].FastGetSolutionStepValue(IS_BOUNDARY) != 0.0)
 		  {
-		    is_free += neighb[i].FastGetSolutionStepValue(IS_FREE_SURFACE);
-		    is_struct += neighb[i].FastGetSolutionStepValue(IS_STRUCTURE);
-		    is_lagin += neighb[i].FastGetSolutionStepValue(IS_LAGRANGIAN_INLET);
+		    if (neighb[i].FastGetSolutionStepValue(IS_FREE_SURFACE) != 0.0)
```

Comment:
Please use flags instead

Reply:
Greeting dear @loumalouomega and very good day to you,

I just discussed this item with Dr. Alex, we can use 

KRATOS_CREATE_FLAG(STRUCTURE, 63);
 
but we need two more variables one for the triple points, and one for the Lagrangian-inlet

"If you need a new one you need to propose in an independent PR. "
so shall we go ahead with this pull request as is so that the code will be compiled, and then we start working on it in an independent PR!

thanks and best regard,

Reply:
There is already one called INLET, the number of flags is limited and maybe we can not have all the ones one would like

Reply:
thanks dear, so we will be needing one for the "triple_point".

best regards,

Reply:
shall we go ahead with this pull request and then create a new pull request, 

or shall we include it in this pull request.

best regards,

Reply:
Yes, change the ones that you can currently change with the existing flags

Reply:
@loumalouomega I am strongly against doing this in this pull request. Flags you are talking about involve changes in 75% of our application and also the meshers we use from MeshingApplication. I suggest to make a separate pull request for "moving" all our old flag variables (IS_FREE_SURFACE, IS_STRUCTURE, IS_TRIPPLE_POINT) to new format. But lets do it in a separate pull request, otherwise things will get mixed.

Reply:
I understand, I have approved

Reply:
I agree with not using flags in this PR

nevertheless I recommend to NOT compare doubles with `==` but rather sth like
`if abs(val1-val2) < 1e-15` (just an example)

Quoting https://floating-point-gui.de/errors/comparison/:
>Due to rounding errors, most floating-point numbers end up being slightly imprecise. As long as this imprecision stays small, it can usually be ignored. However, it also means that numbers expected to be equal (e.g. when calculating the same result through different correct methods) often differ slightly, and a simple equality test fails

I will remove my blocker since my concerns were addressed